### PR TITLE
Telegram extend

### PR DESCRIPTION
**Pull request by skylinezum**
- extend telegram for other methods
- change telegram-parse-spec

### CODE REVIEW

**Comment by stojanovic** [comment]
@skylinezum hey, it seems tests are failing, can you please check why? I don't see anything useful in travis message.

**Comment by skylinezum** [comment]
Hi @stojanovic I asked @gojko and he said that jasmine reporter is updated, so I will update from master and try to get the reporter to show..

**Comment by stojanovic** [comment]
Thanks @skylinezum :)

File: lib/telegram/parse.js [hunk 1]
```diff
@@ -21,4 +21,9 @@ module.exports = function(messageObject) {
       type: 'telegram'
     };
   }
+
+  return {
+    originalRequest: messageObject,
+    type: 'telegram'
+  };
```
Comment (stojanovic) [inline lib/telegram/parse.js]: I think we should still have sender and text, just to keep the standard format. It's ok if text is empty, we already have that for Slack [here](https://github.com/skylinezum/claudia-bot-builder/blob/master/lib/slack/parse.js#L15).

File: lib/telegram/reply.js [hunk 2]
```diff
@@ -4,22 +4,38 @@ const rp = require('minimal-request-promise');
 
 module.exports = function tlReply(messageObject, message, tlAccessToken) {
 
+  /*response from bot
```
Comment (stojanovic) [inline lib/telegram/reply.js]: Let's remove this comment, it's not really useful :)

**Comment by simalexan** [comment]
@skylinezum just one additional, did you try this out locally, on all three types of messages?

**Comment by simalexan** [comment]
@skylinezum could you also add a test for the case you added?

**Comment by gojko** [comment]
To keep the response structure consistent with the other bots, I propose that instead of packing method and other properties together, we allow the users to reply with an object including method and body, so if the response is a string we do special processing, but if the response is an object, we just use the body subkey, instead of copying and filtering the method parameter to get the body

**Comment by skylinezum** [comment]
@gojko Yea, I think having a separate method key and body key makes sense. I'll make changes when I get the chance. @stojanovic Ok I'll put a blank text key and keep them the same. I'll also add tests. I've tested it for what I'm using, but when I have time I'll make the tests better. Thanks for the feedback.

**Comment by stojanovic** [comment]
@skylinezum hey, I'll close this PR because this will be on NPM later today.
Here's what I did:

1. If you send just text it'll send text message via the same message
2. If you send something like `{ text: 'something', parse_mode: 'Markdown' }` it'll send that custom message via the same method, it'll add chat id automatically if you don't pass it
3. If you send something like `{ method: 'sendMessage', body: { text: '...' } }` it'll overwrite the method and apply the same for body as in case 2.

Hope that works :)

**Comment by skylinezum** [comment]
Yea, that works.
Thanks!

**Comment by stojanovic** [comment]
@skylinezum https://github.com/claudiajs/claudia-bot-builder/releases/tag/v1.4.5

**Pull request closed without merging**